test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -4283,24 +4271,34 @@ def create_pbsnode(self, node_name, attrs):
                             'bin', 'qmgr')
         execcmd = "create node " + node_name
         cmd = [qmgr, "-c", execcmd]
-        self.du.run_cmd(
-            self.hostname, cmd, sudo=True)
-        skip_atb_list = ['id', 'Port', 'pbs_version', 'pcpus',
-                         'state', 'last_state_change_time', 'ntype',
+        ret = self.du.run_cmd(self.hostname, cmd, sudo=True)
+        if ret['rc'] != 0:
+            self.logger.info("Failed to create node: %s" % node_name)
+            self.logger.error("Error: %s" % ret['err'])
+            return False
+        # skip all read-only attributes
+        skip_atb_list = ['id', 'pbs_version', 'pcpus',
+                         'last_state_change_time', 'ntype',
                          'Mom', 'sharing', 'resources_available.vnode',
-                         'resources_available.host', 'last_used_time'
+                         'resources_available.host', 'last_used_time',
+                         'resource_assigned', 'resv'
                          ]
         for node_atb, val in attrs.items():
+            # only offline state of node is read, write attribute
             if(node_atb in skip_atb_list or
-               'resources_assigned' in node_atb):
+               'resources_assigned' in node_atb or
+               (node_atb == 'state' and val != 'offline')):
                 continue
             k = str(node_atb)
             v = str(val)
             execcmd = "set node %s %s=%s" % (node_name, k, v)
             cmd = [qmgr, "-c", execcmd]
-            self.du.run_cmd(
-                self.hostname, cmd, sudo=True,
-                logerr=False, level=logging.DEBUG)
+            ret = self.du.run_cmd(self.hostname, cmd, sudo=True,
+                                  logerr=False, level=logging.DEBUG)
+            if ret['rc'] != 0:
+                self.logger.info("Failed to set node attribute %s=%s" % (k, v))
+                self.logger.error("Error: %s" % ret['err'])
```

Comment:
shouldn't you return False here?

Reply:
Set node fails even if i try to set Port on mom, but i have to load Port on node to care of condition: multi mom on single host. That is the reason I am not returning False here

Reply:
Changed